diff --git a/hw/arm/Kconfig b/hw/arm/Kconfig
index 4e50fb1..c8fa84d 100644
--- a/hw/arm/Kconfig
+++ b/hw/arm/Kconfig
@@ -217,6 +217,20 @@ config STELLARIS
     select STELLARIS_GPTM # general purpose timer module
     select UNIMP
 
+config RP2040_SOC
+    bool
+    depends on TCG && ARM
+    select ARM_V7M
+    select PL011
+    select RP2040_SIO
+    select UNIMP
+
+config RASPBERRYPI_PICO
+    bool
+    default y
+    depends on TCG && ARM
+    select RP2040_SOC
+
 config STM32VLDISCOVERY
     bool
     default y
diff --git a/hw/arm/meson.build b/hw/arm/meson.build
index b187b94..4d5ff72 100644
--- a/hw/arm/meson.build
+++ b/hw/arm/meson.build
@@ -100,6 +100,9 @@ arm_common_ss.add(when: 'CONFIG_MAX78000FTHR', if_true: files('max78000fthr.c'))
 arm_common_ss.add(when: 'CONFIG_NETDUINO2', if_true: files('netduino2.c'))
 arm_common_ss.add(when: 'CONFIG_RASPI', if_true: files('bcm2835_peripherals.c'))
 arm_common_ss.add(when: 'CONFIG_RASPI', if_true: files('bcm2838_peripherals.c'))
+arm_common_ss.add(when: 'CONFIG_RP2040_SOC', if_true: files('rp2040.c'))
+arm_common_ss.add(when: 'CONFIG_RASPBERRYPI_PICO',
+                   if_true: files('raspberrypi-pico.c'))
 arm_common_ss.add(when: 'CONFIG_STRONGARM', if_true: files('strongarm.c'))
 arm_common_ss.add(when: 'CONFIG_SX1', if_true: files('omap_sx1.c'))
 arm_common_ss.add(when: 'CONFIG_VERSATILE', if_true: files('versatilepb.c'))
